lib/Doctrine/DBAL/Types/DateIntervalType.php

```diff
@@ -56,9 +58,13 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
         }
 
         try {
-            return new \DateInterval($value);
+            $interval = new \DateInterval(substr($value, 1));
```

Comment:
My 2 cents on this. As already pointed out, this is a BC break that cause schema changes and format changes, which we cannot accept for a minor release. We could however introduce a new type like `SignedDateIntervalType` to keep BC.